### ISSUE TITLE
Handle displaying useful log messages when running `serve`

### ISSUE THREAD

**Issue by vividviolet** [Shopify/shopify-cli-extensions#92]
Currently `admin-ui-extensions-run` and `checkout-ui-extensions-run` both output additional instructions on next steps after running serve. 

We need to ensure the new dev server continues to support this.

**Comment by t6d** [comment]
Couldn't it always be `localhost` and `/extensions` redirects either to another `localhost` URL or the public URL if provided?

**Comment by vividviolet** [comment]
Yes, we can do that but we'd have to change our existing code. Currently we're doing this:
- on `serve` output the server root url based on whether public url was provided
- when accessing the extension root, return a tunnel error if it was accessed via local host

Currently the api package has no knowledge of the public url so we'd need to pass that down for it all to work

**Comment by t6d** [comment]
I'm happy to retain our current approach then as I don't see any benefits from changing it to `localhost`.

**Comment by t6d** [comment]
After some discussion, we decided that the Go server is the best system component for storing and outputting the next step message as it manages all the templates. This ensures that the `shopify-cli-extensions` JS package remains use case agnostic.
